core-bundle/tests/Filesystem/FilesystemItemTest.php

```diff
@@ -196,6 +197,24 @@ static function () use (&$invocationCounts): string {
         }
     }
 
+    public function testThrowsIfMimeTypeIsNotDefined(): void
+    {
+        $item = new FilesystemItem(
+            true,
+            'some/file.txt',
+            null,
+            null,
+            static function (): void {
+                throw VirtualFilesystemException::unableToRetrieveMetadata('some/file.txt');
+            }
+        );
+
+        $this->expectException(VirtualFilesystemException::class);
```

Comment:
Afaik we usually put the expected exception calls first in the tests :)

Reply:
No, you are confusing this with `$this->expectDeprecation()`. The position of the `$this->expectException()` call is not arbitrary and should be as close to the call triggering the exception as possible.